### PR TITLE
Issue #2827: JavadocMethod doesn't report anonymous class methods

### PR DESCRIPTION
**Pull request by AayushSaini-Temp**
Related: https://github.com/checkstyle/checkstyle/issues/2827
Diff Regression config: https://gist.githubusercontent.com/AayushSaini101/218a9a57a04728fb154b406078059651/raw/9122a6c84272be78546541c4adf7ee3135d99b99/gistfile1.txt

Diff Regression Project: https://gist.githubusercontent.com/AayushSaini101/0747edfd032791a6db97ab895e5c930c/raw/2b6a38b42b347c8827b7423cdd6086e40d95a236/gistfile1.txt

Diff Regression patch config: https://gist.githubusercontent.com/AayushSaini-Temp/084f946327fa1e4e5781b333df8e7137/raw/38d12b9e159aa8abd51d08d2c650818fc6584d94/gistfile1.txt

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java [hunk 1]
```diff
@@ -293,6 +303,16 @@ public void setValidateThrows(boolean value) {
         validateThrows = value;
     }
 
+    /**
+     * Setter to control whether to validateInAnonymousClasses.
+     *
+     * @param value user's value.
+     * @since 6.0
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java:310]: Put 10.13.0, it will be next release number

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java (hunk 1 above)
Comment (AayushSaini-Temp) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java]: Done @romani

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodValidateAnonymousClass.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+*/
+
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocmethod;
+
+public class InputJavadocMethodValidateAnonymousClass {
+
+     Thread ob = new Thread(){
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodValidateAnonymousClass.java:19]: Please add more examples of anonymous classes usages.
You can copy ideas from diff report.
Let's have anon classes in ~10 different places: in method, in expression, as parameters,....

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodValidateAnonymousClass.java (hunk 2 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodValidateAnonymousClass.java:19]: @romani Can you please tell me how can i use the diff  report for the ideas

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodValidateAnonymousClass.java:19]: In PR https://github.com/checkstyle/checkstyle/pull/13968

You have report https://github.com/checkstyle/checkstyle/pull/13968#issuecomment-1788620332
It has a lot of differences, anonymous classes in many areas of code.
But in your PR you have only single test. Yes, is is minimal required test, but we should not depend on diff report too much, we should have in our code base set of Inputs that has targeting code in almost all possible positions.

In short: copy code to inputs from diff report to have anonymous class in not only in field of class.

**Comment by romani** [comment]
As you introduce new property you need to use special config for your codebase execution.
`Diff Regression patch config:`
https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation

**Comment by aayushRedHat** [comment]
@romani Please consider this PR https://github.com/checkstyle/checkstyle/pull/13992 some issue with cLI
